Source/WTF/wtf/WTFConfig.cpp

```diff
@@ -109,12 +109,12 @@ void Config::permanentlyFreeze()
 #if ENABLE(UNIFIED_AND_FREEZABLE_CONFIG_RECORD)
 #if PLATFORM(COCOA)
     enum {
-        AllowPermissionChangesAfterThis = false,
-        DisallowPermissionChangesAfterThis = true
+        DontUpdateMaximumPermission = false,
```

Comment:
Should this be UpdateCurrentPermissionOnly? I am confused about the docs

Reply:
Nope.  Prototype of `vm_protect` is:
```
kern_return_t vm_protect
(
	vm_map_t target_task,
	vm_address_t address,
	vm_size_t size,
	boolean_t set_maximum,
	vm_prot_t new_protection
);
```

`DontUpdateMaximumPermission` is accurate because it tells `vm_protect` to restrict permissions to what is specified (just `VM_PROT_READ` in this case), and to not allow permissions to ever be set to anything less restrictive after this (e.g. adding back write permissions: the maximum permission is now only read).